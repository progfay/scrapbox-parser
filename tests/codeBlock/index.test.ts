/* global describe it expect */

import '../jest-setup'

describe('Code Block', () => {
  it('Simple code block', () => {
    expect(`code:hello.js
 function () {
   alert(document.location.href)
   console.log("hello")
   // You can also write comments!
 }`).toEqualWhenParsing([
      {
        indent: 0,
        type: 'codeBlock',
        fileName: 'hello.js',
        content: 'function () {\n  alert(document.location.href)\n  console.log("hello")\n  // You can also write comments!\n}'
      }
    ])
  })

  it('Bulleted code block', () => {
    expect(` code:hello.js
  function () {
    alert(document.location.href)
    console.log("hello")
    // You can also write comments!
  }`).toEqualWhenParsing([
      {
        indent: 1,
        type: 'codeBlock',
        fileName: 'hello.js',
        content: 'function () {\n  alert(document.location.href)\n  console.log("hello")\n  // You can also write comments!\n}'
      }
    ])
  })

  it('Code block with bullet', () => {
    expect(` Bullet
 code:hello.js
  function () {
    alert(document.location.href)
    console.log("hello")
    // You can also write comments!
  }
 Bullet`).toEqualWhenParsing([
      {
        indent: 1,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: 'Bullet'
          }
        ]
      },
      {
        indent: 1,
        type: 'codeBlock',
        fileName: 'hello.js',
        content: 'function () {\n  alert(document.location.href)\n  console.log("hello")\n  // You can also write comments!\n}'
      },
      {
        indent: 1,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: 'Bullet'
          }
        ]
      }
    ])
  })

  it('Consecutive code blocks', () => {
    expect(`code:hello.js
 function () {
   alert(document.location.href)
   console.log("hello")
   // You can also write comments!
 }
code:hello.js
 function () {
   alert(document.location.href)
   console.log("hello")
   // You can also write comments!
 }`).toEqualWhenParsing([
      {
        indent: 0,
        type: 'codeBlock',
        fileName: 'hello.js',
        content: 'function () {\n  alert(document.location.href)\n  console.log("hello")\n  // You can also write comments!\n}'
      },
      {
        indent: 0,
        type: 'codeBlock',
        fileName: 'hello.js',
        content: 'function () {\n  alert(document.location.href)\n  console.log("hello")\n  // You can also write comments!\n}'
      }
    ])
  })
})
