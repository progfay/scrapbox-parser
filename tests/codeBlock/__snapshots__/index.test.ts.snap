// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Block Bulleted code block 1`] = `
[
  {
    "content": "function () {
  alert(document.location.href)
  console.log("hello")
  // You can also write comments!
}",
    "fileName": "hello.js",
    "indent": 1,
    "type": "codeBlock",
  },
]
`;

exports[`Code Block Code block with bullet 1`] = `
[
  {
    "indent": 1,
    "nodes": [
      {
        "raw": "Bullet",
        "text": "Bullet",
        "type": "plain",
      },
    ],
    "type": "line",
  },
  {
    "content": "function () {
  alert(document.location.href)
  console.log("hello")
  // You can also write comments!
}",
    "fileName": "hello.js",
    "indent": 1,
    "type": "codeBlock",
  },
  {
    "indent": 1,
    "nodes": [
      {
        "raw": "Bullet",
        "text": "Bullet",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`Code Block Consecutive code blocks 1`] = `
[
  {
    "content": "function () {
  alert(document.location.href)
  console.log("hello")
  // You can also write comments!
}",
    "fileName": "hello.js",
    "indent": 0,
    "type": "codeBlock",
  },
  {
    "content": "function () {
  alert(document.location.href)
  console.log("hello")
  // You can also write comments!
}",
    "fileName": "hello.js",
    "indent": 0,
    "type": "codeBlock",
  },
]
`;

exports[`Code Block Simple code block 1`] = `
[
  {
    "content": "function () {
  alert(document.location.href)
  console.log("hello")
  // You can also write comments!
}",
    "fileName": "hello.js",
    "indent": 0,
    "type": "codeBlock",
  },
]
`;
